### PR TITLE
PLANET-3206 Fix empty string supplied PHP warning in Submenu block

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**

### CODE REVIEW

File: classes/controller/blocks/class-submenu-controller.php [hunk 1]
```diff
@@ -217,7 +217,9 @@ private function parse_post_content( $content, $attributes ) {
 			}
 
 			$dom = new \DOMDocument();
-			$dom->loadHtml( $content );
+			if ( '' === $content ) {
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: If I am not mistaken we should make sure that $content is not empty.
We could do something like
`if ( ! $content ) { return []; }`

**Comment by angtheod** [comment]
I get a couple more warnings when I run `wp elasticpress index`.

One is in submenu block again [here](https://github.com/greenpeace/planet4-plugin-blocks/blob/release/v1.33.0/classes/controller/blocks/class-submenu-controller.php#L184) and we need to check if $post is an object like this:
`$content = $post instanceof \WP_Post ? $post->post_content : $content;`

the other one is in engagingnetworks plugin [here](https://github.com/greenpeace/planet4-plugin-engagingnetworks/blob/develop/classes/controller/blocks/class-enform-controller.php#L350) and we need to check that array has at least 2 elements like this:
`if ( $attr_parts_num >= 2 && is_numeric( $attr_parts[0] ) && 'mandatory' !== $attr_parts[ $attr_parts_num - 1 ] ) {`

File: classes/controller/blocks/class-submenu-controller.php (hunk 1 above)
Comment (kirdia) [reply classes/controller/blocks/class-submenu-controller.php]: Yes, I agree with @angelos on this. No need to continue execution if no content is set.
Returning an empty array at this point would be enough.

Comment (angelos) [reply classes/controller/blocks/class-submenu-controller.php]: Thanks for including me in the conversation! However, I'm pretty sure you meant to mention @angtheod :)

Comment (kirdia) [reply classes/controller/blocks/class-submenu-controller.php]: hahaha :smile:  yes, true. it was meant for @angtheod. sorry for the false notification :smiley:

**Comment by angtheod** [comment]
Please fix also the PHPCS error.

**Pull request merged**